drivers/gpio/gpio_pinctrl_stm32.c

```diff
@@ -0,0 +1,685 @@
+/*
+ * Copyright (c) Bobby Noelte
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Keep this first */
+#include "gpio_pinctrl_stm32.h"
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/types.h>
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <device.h>
+#include <soc.h>
+#include <pinctrl.h>
+#include <gpio.h>
+
+#include <chaos/preprocessor/cat.h>
+#include <chaos/preprocessor/control/when.h>
+#include <chaos/preprocessor/repetition/repeat_from_to.h>
+#include <chaos/preprocessor/array/elem.h>
+#include <chaos/preprocessor/string/to_lower.h>
+#include <chaos/preprocessor/detection/is_numeric.h>
+
+#include <clock_control/stm32_clock_control.h>
+#include <interrupt_controller/exti_stm32.h>
+
+#include "gpio_utils.h" /* _gpio_manage_callback, _gpio_fire_callbacks */
+
+/**
+ * @brief Common GPIO driver for STM32 MCUs.
+ *
+ * Uses the PINCTRL driver as a backend driver.
+ */
+
+/**
+ * @def GPIO_PINCTRL_STM32_PORTS_COUNT
+ * @brief Number of GPIO ports to be managed by the GPIO drivers.
+ */
+/* GPIOx is defined by the STM32 CubeMX library. */
+#if defined(GPIOK)
+#define GPIO_PINCTRL_STM32_PORTS_COUNT 11
+#elif defined(GPIOJ)
+#define GPIO_PINCTRL_STM32_PORTS_COUNT 10
+#elif defined(GPIOI)
+#define GPIO_PINCTRL_STM32_PORTS_COUNT 9
+#elif defined(GPIOH)
+#define GPIO_PINCTRL_STM32_PORTS_COUNT 8
+#elif defined(GPIOG)
+#define GPIO_PINCTRL_STM32_PORTS_COUNT 7
+#elif defined(GPIOF)
+#define GPIO_PINCTRL_STM32_PORTS_COUNT 6
```

Comment:
As mentioned for PINCTRL, in some SoCs, GPIOH can be present while GPIOF isn't.

Reply:
No Problem as long as you do not activate the node in the device tree, the device will not be created.

Reply:
ok, I'll have a try and see how this all works.